lib/rubocop/cop/rspec/factory_bot/attribute_defined_statically.rb

```diff
@@ -72,14 +72,7 @@ def proc?(attribute)
             value_matcher(attribute).to_a.all?(&:block_pass_type?)
           end
 
-          def association?(attribute)
-            argument = attribute.first_argument
-            argument.hash_type? && factory_key?(argument)
-          end
-
-          def factory_key?(hash_node)
-            hash_node.keys.any? { |key| key.sym_type? && key.value == :factory }
-          end
+          def_node_matcher :association?, '(hash <(pair (sym :factory) _) ...>)'
```

Comment:
❤️ It’s so concise! I just hope people (we) will find it readable in the future.